src/js/view/controls/components/selection-display-menu.js

```diff
@@ -29,16 +29,15 @@ define([
         selectItem(selectedIndex) {
             super.selectItem(selectedIndex);
 
-            if (selectedIndex !== -1) {
-                var isDefaultOption = selectedIndex === this.defaultIndex;
+            // Show the default icon if we have the default value selected, or one that isn't in the menu
+            var isDefaultOption = selectedIndex === this.defaultIndex || selectedIndex === -1;
```

Comment:
Let's write this so the condition answers the question: should we show a text label